R/derive_extreme_event.R

```diff
@@ -486,6 +482,234 @@ derive_extreme_event <- function(dataset,
+#'       the number of the event.
+#'   1. All selected observations are bound together.
+#'   1. For each group (with respect to the variables specified for the
+#'   `by_vars` parameter) the first or last observation (with respect to the
+#'   order specified for the `order` parameter and the mode specified for the
+#'   `mode` parameter) is selected.
+#'   1. The variables specified by the `new_vars` parameter are added to
+#'   the selected observations.
+#'   1. The variables are added to input dataset.
+#'
+#'
+#' @return The input dataset with the best or worst observation of each by group
+#'   added as new variables.
+#'
+#' @family der_adsl
+#' @keywords der_adsl
+#'
+#' @seealso [event()], [event_joined()], [derive_extreme_event()]
+#'
+#' @export
+#'
+#' @examples
+#' library(tibble)
+#' library(dplyr)
+#' library(lubridate)
+#'
+#' adsl <- tribble(
+#'   ~STUDYID, ~USUBJID, ~TRTEDT, ~DTHDT,
+#'   "PILOT01", "01-1130", ymd("2014-08-16"), ymd("2014-09-13"),
+#'   "PILOT01", "01-1133", ymd("2013-04-28"), ymd(""),
+#'   "PILOT01", "01-1211", ymd("2013-01-12"), ymd(""),
+#'   "PILOT01", "09-1081", ymd("2014-04-27"), ymd(""),
+#'   "PILOT01", "09-1088", ymd("2014-10-09"), ymd("2014-11-01"),
+#' )
+#'
+#' lb <- tribble(
+#'   ~STUDYID,  ~DOMAIN,  ~USUBJID, ~LBSEQ,             ~LBDTC,
+#'   "PILOT01",    "LB", "01-1130",    219, "2014-06-07T13:20",
+#'   "PILOT01",    "LB", "01-1130",    322, "2014-08-16T13:10",
+#'   "PILOT01",    "LB", "01-1133",    268, "2013-04-18T15:30",
+#'   "PILOT01",    "LB", "01-1133",    304, "2013-05-01T10:13",
+#'   "PILOT01",    "LB", "01-1211",      8, "2012-10-30T14:26",
+#'   "PILOT01",    "LB", "01-1211",    162, "2013-01-08T12:13",
+#'   "PILOT01",    "LB", "09-1081",     47, "2014-02-01T10:55",
+#'   "PILOT01",    "LB", "09-1081",    219, "2014-05-10T11:15",
+#'   "PILOT01",    "LB", "09-1088",    283, "2014-09-27T12:13",
+#'   "PILOT01",    "LB", "09-1088",    322, "2014-10-09T13:25"
+#' ) %>%
+#'   mutate(
+#'     ADT = convert_dtc_to_dt(LBDTC)
+#'   )
+#'
+#' derive_vars_extreme_event(
+#'   adsl,
+#'   by_vars = exprs(STUDYID, USUBJID),
+#'   events = list(
+#'     event(
+#'       dataset_name = "adsl",
+#'       condition = !is.na(DTHDT),
+#'       set_values_to = exprs(LSTALVDT = DTHDT, DTHFL = "Y")
+#'     ),
+#'     event(
+#'       dataset_name = "lb",
+#'       condition = !is.na(ADT),
+#'       order = exprs(ADT),
+#'       mode = "last",
+#'       set_values_to = exprs(LSTALVDT = ADT, DTHFL = "N")
+#'     ),
+#'     event(
+#'       dataset_name = "adsl",
+#'       condition = !is.na(TRTEDT),
+#'       order = exprs(TRTEDT),
+#'       mode = "last",
+#'       set_values_to = exprs(LSTALVDT = TRTEDT, DTHFL = "N")
+#'     )
+#'   ),
+#'   source_datasets = list(adsl = adsl, lb = lb),
+#'   tmp_event_nr_var = event_nr,
+#'   order = exprs(LSTALVDT, event_nr),
+#'   mode = "last",
+#'   new_vars = exprs(LSTALVDT = LSTALVDT, DTHFL = DTHFL)
```

Comment:
Can be simplified to `new_vars = exprs(LSTALVDT, DTHFL)`.

Reply:
Updated, thanks!